src/main.rs

```diff
@@ -149,57 +152,59 @@ async fn run(config: Option<String>, noinit: bool) -> Result<()> {
         let message_sender = message_sender.clone();
         let id = blocks_tasks.len();
 
-        blocks_tasks.push(tokio::spawn(run_block(
+        blocks_tasks.push(run_block(
```

Comment:
@remi-dupre Now that everything runs on one thread, we can just use [FuturesUnordered](https://docs.rs/futures/0.3.14/futures/stream/struct.FuturesUnordered.html) directly without `spawn_local()`, right?